src/ray/core_worker/core_worker.cc

```diff
@@ -248,6 +260,19 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
   }
 }
 
+void CoreWorker::HandleWorkerFailed(const WorkerID &worker_id,
+                                    const gcs::WorkerFailureData &worker_failed_data) {
+  RAY_LOG(DEBUG) << "Worker " << worker_id << " failed";
+  // TODO: Clean up after the failure:
+  // - If the failed worker is our owner, then exit.
```

Comment:
Could we feature-flag this? It's not related to the current PR though:)

Reply:
I don't think it makes sense to feature-flag since it's absolutely necessary for stability. Otherwise, you could end up with a resource leak after a worker failure.

Reply:
Right, I also meant fate sharing with the owner. It does not make sense to feature-flag because it is always desirable as a default. Of course, for cases like detached actors, the actor would not fate-share with the owner. In the future, we may need to add other such special cases that can be specified by the user code, but we should not feature-flag it in the backend.